runtime/connectors/connectors.go

```diff
@@ -95,6 +97,12 @@ type SamplePolicy struct {
 	Limit    int
 }
 
+type Iterator interface {
+	Close() error
+	NextBatch(ctx context.Context, n int) ([]string, error)
+	HasNext() bool
+}
+
```

Comment:
1. Makes sense. Will add docstring.
2. I don't agree with returning an error on done. A lot of languages do expose a separate function to check if iterator has more elements left. This removes the necessity to add a separate check for error. Also fits well in for loop constructs. Agree on renaming it to file iterator.
3. I did think about it. Didn't have any strong opinion about keeping/not keeping in iterator. But when I think it now I think it will reduce the number of arguments being passed to ctx so will keep this in iterator state itself.